src/basic/fd-util.h

```diff
@@ -126,3 +127,12 @@ static inline char *format_proc_fd_path(char buf[static PROC_FD_PATH_MAX], int f
 
 #define FORMAT_PROC_FD_PATH(fd) \
         format_proc_fd_path((char[PROC_FD_PATH_MAX]) {}, (fd))
+
+static inline const char *format_fd_path(char buf[static PATH_MAX], int fd) {
+        buf[0] = 0;
+        (void) readlink(FORMAT_PROC_FD_PATH(fd), buf, PATH_MAX);
+        return buf;
+}
```

Comment:
i think adding this is OK, as long as it is used for logging only. Can you add a comment that underlines that, and tells people to not use it for anything bug logging?